packages/@aws-cdk/aws-batch/lib/compute-environment.ts

```diff
@@ -0,0 +1,333 @@
+import ec2 = require('@aws-cdk/aws-ec2');
+import iam = require('@aws-cdk/aws-iam');
+import { Construct, IResource, Resource, Stack, Tag } from '@aws-cdk/core';
+import { CfnComputeEnvironment } from './batch.generated';
+
+export enum ComputeEnvironmentType {
+    MANAGED   = 'MANAGED',
+    UNMANAGED = 'UNMANAGED',
+}
+
+export enum ComputeResourceType {
+    EC2  = 'EC2',
+    SPOT = 'SPOT',
+}
+
+export enum AllocationStrategy {
+    BEST_FIT = 'BEST_FIT',
+    BEST_FIT_PROGRESSIVE = 'BEST_FIT_PROGRESSIVE',
+    SPOT_CAPACITY_OPTIMIZED = 'SPOT_CAPACITY_OPTIMIZED',
+}
+
+export interface ComputeResourceProps {
+    /**
+     * The IAM role applied to EC2 resources in the compute environment.
+     */
+    readonly instanceRole: iam.IRole;
+
+    /**
+     * The types of EC2 instances that may be launched in the compute environment. You can specify instance
+     * families to launch any instance type within those families (for example, c4 or p3), or you can specify
+     * specific sizes within a family (such as c4.8xlarge). You can also choose optimal to pick instance types
+     * (from the C, M, and R instance families) on the fly that match the demand of your job queues.
+     *
+     * @default optimal
+     */
+    readonly instanceTypes?: ec2.InstanceType[];
```

Comment:
A very good question, its because of the `instance types` property type here in the Cloudformation documentation for the [AWS::Batch::ComputeEnvironment](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-batch-computeenvironment-computeresources.html#cfn-batch-computeenvironment-computeresources-instancetypes) resource.